### DISCUSSION TITLE
How to Check If an insert is made successfully.

### DISCUSSION THREAD

**Discussion by rahat14** [simolus3/drift#1701] [Q&A]
I just want to check if an insert or update ran successfully or not. 
//database code 
`  Future insertProductInRunningCart(RunningCartProductCompanion item ) => into(runningCartProduct).insert(
      item, mode: InsertMode.insertOrReplace
  );
`
// calling in controller
`database.insertProductInRunningCart(OfflineItem);`

now how can I check for any error or determine that this insertion ran successfully? .  Please help.

**Comment by simolus3** [comment]
If the future doesn't complete with an error, the insertion ran successfully. You can check for errors by awaiting `insertProductInRunningCart` in a try/catch block or with `.onError` / the other future error handling mechanisms.
